src/mapreduce.jl

```diff
@@ -90,6 +90,36 @@ end
     end
 end
 
+@generated function reduce{D}(op, a::StaticArray, ::Type{Val{D}})
```

Comment:
If a single dimensionis is sufficient, that will do for now.

One way of doing this is `reducedim(op, a, Val{(D1,D2,...)})` so later we can check if `D` is an integer or a tuple and expand the tuple. Alternatively they could be applied in sequence, like `reducedim(op, a, Val{D1}, Val{D2}, ...)`. 

I don't think this is necessary for this PR however.